lib/refinement/cocoapods_post_install_writer.rb

```diff
@@ -0,0 +1,118 @@
+module Refinement
+  # Called after CocoaPods installation to write an augmenting file that
+  # takes into account changes to Pod configuration,
+  # as well as the globs used by podspecs to search for files
+  class CocoaPodsPostInstallWriter
+    attr_reader :aggregate_targets, :config, :repo, :options
+    private :aggregate_targets, :config, :repo, :options
+
+    # Initializes a post-install writer with CocoaPods target objects.
+    # @return [CocoaPodsPostInstallWriter] a new instance of CocoaPodsPostInstallWriter
+    # @param aggregate_targets [Array<Pod::AggregateTarget>]
+    # @param config [Pod::Config]
+    # @param options [Hash]
+    def initialize(aggregate_targets, config, options)
+      @aggregate_targets = aggregate_targets
+      @config = config
+      @repo = config.installation_root
+      @options = options || {}
+    end
+
+    # Writes the refinement augmenting file to the configured path
+    # @return [Void]
+    def write!
+      write_file options.fetch('output_path', config.sandbox.root.join('pods_refinement.json')), paths_by_target_name
+    end
+
+    private
+
+    def write_file(path, hash)
+      require 'json'
+      File.open(path, 'w') do |f|
+        f << JSON.generate(hash) << "\n"
+      end
+    end
+
+    def paths_by_target_name
+      targets = {}
+      aggregate_targets.each do |aggregate_target|
+        targets[aggregate_target.label] = paths_for_aggregate_target(aggregate_target)
+      end
+      aggregate_targets.flat_map(&:pod_targets).uniq.each do |pod_target|
+        targets.merge! paths_for_pod_targets(pod_target)
```

Comment:
Is running `merge!` sufficient? Or is there some kind of deeper merge we would have to run in order to preserve all the added paths with inclusion reasons? I tested locally with this example:
`        2.3.1 :046 > f = { 'a_target':  [ { path: 'a', inclusion_reason: 'its good' } ] }
         => {:a_target=>[{:path=>"a", :inclusion_reason=>"its good"}]}
        2.3.1 :047 > g =  { 'a_target':  [ { path: 'a', inclusion_reason: 'its good' }, { path: 'b',  inclusion_reason: 'its great' } ] }
         => {:a_target=>[{:path=>"a", :inclusion_reason=>"its good"}, {:path=>"b", :inclusion_reason=>"its great"}]}
        2.3.1 :048 > h =  { 'a_target':  [ { path: 'a', inclusion_reason: 'its cool' }, { path: 'c',  inclusion_reason: 'its great' } ] }
         => {:a_target=>[{:path=>"a", :inclusion_reason=>"its cool"}, {:path=>"c", :inclusion_reason=>"its great"}]}
        2.3.1 :049 > f.merge(g).merge(h)
         => {:a_target=>[{:path=>"a", :inclusion_reason=>"its cool"}, {:path=>"c", :inclusion_reason=>"its great"}]}
        #  note that path B gets clobbered`

Reply:
Running `merge!` is sufficent, since no two pod targets in the same installation can have the same `label`, which is the key in the dictionary we're merging in